agents/builder/Dockerfile

```diff
@@ -0,0 +1,22 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
What's this Dockerfile for?

Reply:
Being a container that can pull the contents of a user's directory that was staged to GCS, build a container from that, and push that container to a registry that can be accessed for the initialization of the TfJob.
The command that is run is here https://github.com/cwbeitel/examples/blob/master/agents/config/builder.yaml#L28

This approach could be replaced by using Google Container Builder in the short term and simplify things.

In general it seems important for there to be a vendor-agnostic container build step that feeds into training. But that doesn't mean there needs to be a separate builder container. Docker can simply be installed in the user's container and these gsutil and gcloud operations should probably be python API calls instead of command line operations!